src/main/java/seedu/duke/lesson/Lesson.java

```diff
@@ -38,4 +59,43 @@ public void setTeachingStaff(TeachingStaff teachingStaff) {
     public void setOnlineLink(String onlineLink) {
         this.onlineLink = onlineLink;
     }
+
+    public LessonType initialiseLessonType(String lessonType) {
+        switch (lessonType) {
+        case TUTORIAL:
+            return LessonType.TUTORIAL;
+
+        case LECTURE:
+            return LessonType.LECTURE;
+
+        case LAB:
+            return LessonType.LAB;
+
+        default:
+            //throw required exception if parser didn't handle
+            System.out.println("Wrong name");
+            return LessonType.LECTURE; //temporary
+        }
+    }
+
+    /**
+     * User input: "add tutorial ;; Wednesday 9 am - 10am ;; https://nus-sg.zoom.us/j/abc"
+     * Parser input: "tutorial ;; Wednesday 9 am - 10am ;; https://nus-sg.zoom.us/j/abc ;;  ;; "
+     * Parser will input " " and delimiters in parser input equal to five i,e
+     * Tentatively discussed with Ivan(REMOVE THIS COMMENT AFTER IMPLEMENTATION)
+     * This method add a new lesson to the list of lessons of a module
+     *
+     * @param parserInput is the input obtained from parser assumed to be in the correct format
+     * @param lessonList  is the list of lessons for one module
+     */
+    public static void addNewLesson(String parserInput, ArrayList<Lesson> lessonList) {
+        String[] arguments = getArguments(parserInput);
+        Lesson newLesson = new Lesson(arguments);
+        lessonList.add(newLesson);
+        UI.printAddNewLesson(newLesson.lessonType);
+    }
+
+    private static String[] getArguments(String parserInput) {
+        return parserInput.split(DELIMITER, NUMBER_OF_ARGUMENTS);
+    }
```

Comment:
Will the validity of arguements be checked here? Or will that be done in the parser?

Reply:
Parser